handler/handler.go

```diff
@@ -45,16 +48,20 @@ func Register(r *mux.Router, opt *Options) {
 	plexus.HandleFunc("/config/{id}:{token}", h.GetAgentMsh).Methods(http.MethodGet)
 	plexus.HandleFunc("/agent/{id}:{token}", h.ProxyAgent).Methods(http.MethodGet)
 	plexus.HandleFunc("/meshrelay.ashx", h.ProxyRelay).Methods(http.MethodGet)
+	plexus.HandleFunc("/pairing/{code}", h.Pair).Methods(http.MethodGet)
 	r.PathPrefix(h.ProxyMeshCentralURL()).Handler(h.ProxyMeshCentral())
 }
 
 type Handler struct {
 	log                logger.Logger
-	cfg                *control.Config
+	ccfg               *control.Config
+	pcfg               *pairing.Config
 	auth               AuthChecker
 	lock               sync.RWMutex
+	codeLock           sync.RWMutex
```

Comment:
Use the existing lock instead of codeLock, in almost all cases you also need the normal session lock, thus, no need for two locks. Having to locks just adds the possibility for deadlocks.